lib/src/main/java/io/ably/lib/transport/Hosts.java

```diff
@@ -60,9 +60,19 @@ public Hosts(String primaryHost, String defaultHost, ClientOptions options) thro
             setPrimaryHost(defaultHost);
         }
         fallbackHostsUseDefault = options.fallbackHostsUseDefault;
-        if (options.fallbackHosts == null) {
-            fallbackHosts = Arrays.copyOf(Defaults.HOST_FALLBACKS, Defaults.HOST_FALLBACKS.length);
-            fallbackHostsIsDefault = true;
+        if (options.tlsPort != 0 || options.port != 0) {
+            fallbackHosts = new String[]{};
+            fallbackHostsIsDefault = false;
+        } else if (options.fallbackHosts == null) {
+            if (!fallbackHostsUseDefault && options.environment != null && !options.environment.equalsIgnoreCase("production")) {
+                /* RSC15g2: If ClientOptions#environment is set to a value other than "production"
+                 * and ClientOptions#fallbackHosts is not set, use the environment fallback hosts */
+                fallbackHosts = Defaults.getEnvironmentFallbackHosts(options.environment);
+                fallbackHostsIsDefault = false;
+            } else {
+                fallbackHosts = Arrays.copyOf(Defaults.HOST_FALLBACKS, Defaults.HOST_FALLBACKS.length);
```

Comment:
RSC15b, RTN17b: only use fallback if the hostname has not been overridden.
This might set default fallbackHosts even if a custom host is set. We will not use default fallbackhosts if the custom host is set. This is just my hunch by observing the code. You can check by writing tests.